src/event/sys/unix/waker.rs

```diff
@@ -0,0 +1,87 @@
+use std::sync::{Arc, Mutex};
+
+use mio::{Evented, Poll, PollOpt, Ready, Registration, SetReadiness, Token};
+
+use crate::Result;
+
+struct WakerInner {
+    registration: Registration,
+    set_readiness: SetReadiness,
+}
+
+impl WakerInner {
+    fn new() -> Self {
+        let (registration, set_readiness) = Registration::new2();
+
+        Self {
+            registration,
+            set_readiness,
+        }
+    }
+
+    fn wake(&self) -> Result<()> {
+        self.set_readiness.set_readiness(Ready::readable())?;
```

Comment:
I see that the type is named `SetReadiness` so it makes sence to have it called `set_readiness` tough, `set_readiness.set_readiness()` looks a bit odd.

Reply:
Yeah, agree, but it's temporary till `mio 0.7` & [`Waker`](https://github.com/tokio-rs/mio/blob/master/src/waker.rs).